addons/account_peppol/models/res_company.py

```diff
@@ -191,18 +191,17 @@ def _compute_account_peppol_phone_number(self):
 
     @api.model
     def _sanitize_peppol_endpoint(self, vals, eas=False, endpoint=False):
-        if 'peppol_eas' not in vals and 'peppol_endpoint' not in vals:
-            return vals
+        peppol_eas = vals.get('peppol_eas', eas)
+        peppol_endpoint = vals.get('peppol_endpoint', endpoint)
 
-        peppol_eas = vals['peppol_eas'] if 'peppol_eas' in vals else eas # let users remove the value
-        peppol_endpoint = vals['peppol_endpoint'] if 'peppol_endpoint' in vals else endpoint
         if not peppol_eas or not peppol_endpoint:
             return vals
 
-        if peppol_eas == '0208':
-            cbe_match = re.search('[0-9]{10}', peppol_endpoint)
-            if bool(cbe_match):
-                vals['peppol_endpoint'] = cbe_match.group(0)
+        endpoint_rule = PEPPOL_ENDPOINT_RULES.get(peppol_eas)
```

Comment:
```suggestion
        if (
            (eas_country := PEPPOL_EAS_MAPPING.get(peppol_eas))
            and self.env['res.partner'].simple_vat_check(eas_country, peppol_endpoint)
        ):
            vals['peppol_endpoint'] = peppol_endpoint
```

We already have checks for VAT numbers, in case the EAS is one.
I don't think there's even need to exclude EAN, since if it doesn't recognize the country, it simply returns False.

I'm surprised we only deal with EAS codes from these 4 countries.
We might as well take the time to see the full EAS mapping... ?
    @dbkosky https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/

I don't know if we're dealing with errors in some way. It looks like we only check for Belgium. Yet, the address looks quite a fundamental required information.

Reply:
You also include EAS number that isn't active and should be consider legacy.. By the way the check is already done on the second sweep, see the .code provided below.Tthe cleanup function is just to remove the problem when people put VAT numbers in the peppol endpoint feel, and i can confirm that this is done by the upgrade tool for some strange reason.

```
    def _check_peppol_endpoint_number(self, warning=False):
        self.ensure_one()

        peppol_dict = PEPPOL_ENDPOINT_WARNING if warning else PEPPOL_ENDPOINT_RULES
        endpoint_rule = peppol_dict.get(self.peppol_eas)
        if not endpoint_rule:
            return True

        if endpoint_rule[0] == 'regex':
            return bool(re.match(endpoint_rule[1], self.peppol_endpoint))

        if endpoint_rule[0] == 'ean':
            check_module = ean
        else:
            check_module = get_cc_module(endpoint_rule[0], endpoint_rule[1])
        return check_module.is_valid(self.peppol_endpoint)
```
The get_cc_module will do the VAT, orgnr, you name it check from the stdnum

Reply:
I do think I need to explain a few things here: There is a difference between VAT and organisation number, even if they do look almost the same.

I will write the peppo-id as <eas>:<endpoint>.
If we take Belgium as an example and Odoo as company it looks like this, if you are not using a GLN-number:

Odoo S.A.
VAT: BE0477472701
PEPPOL-ID: 0208:0477472701

The sanitize code is used in case you write the VAT-number in the field. If we take my company as an example:

XCLUDE AB.
VAT: SE559199111101
Org.Nr: 559199-1111
PEPPOL-ID: 0007:5591991111

So the idea is to do the same for the other countries as well to clear off problems as in Belgium, by just refractoring the code and make it better with less work to move it between versions. By the way, I wouldn't use ':=' (assignment operator) in Odoo, since it was introduced in Python 3.8 and if there are installations running older versions of python it will not work.

Here is the full list of EAS-numbers that is active for your reference:

https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/

Reply:
Thanks for the tip, I have missed it totally. Then I will start using assignement operators in my code.